vpr/src/pack/pb_type_graph.cpp

```diff
@@ -349,11 +353,14 @@ static void alloc_and_load_pb_graph(t_pb_graph_node* pb_graph_node,
                                                                                 sizeof(t_pb_graph_node*));
         for (j = 0; j < pb_type->modes[i].num_pb_type_children; j++) {
             pb_graph_node->child_pb_graph_nodes[i][j] = (t_pb_graph_node*)vtr::calloc(pb_type->modes[i].pb_type_children[j].num_pb, sizeof(t_pb_graph_node));
+            int base = flat_index*(pb_type->modes[i].pb_type_children[j].num_pb);
```

Comment:
1. Explain this -- what is this mapping?
2. Should this be a helper function?  It seems this may be a specific mapping function/style so putting it in a function will give it a name (helps readability), give a natural place to comment it, and make it easier to change or generalize the mapping function in the future if that ever becomes necessary.

Reply:
flat_index*(pb_type->modes[i].pb_type_children[j].num_pb) is the flattened index, at the level of the child, up until the parent.  

For example, say there are four adults and each has three kids; then the base value (in terms of kid count) at the third adult is 6 (because the first and second adults each have three kids),. The third adult's kids have indices (6+0), (6+1), and (6+2).  

It would be most readable to call some sort of "get child's flat index" function once per child, but then you'd end up computing the same parent base multiple times; I do it outside the k-loop so it only needs to be computed once. 

How would you like me to handle it?  I think either is fine; this function is not called all that many times, so a slight inefficiency for the sake of legibility is probably ok.